### PR TITLE
NT-1764:Refactor ThanksActivity

### PR DESCRIPTION
**Pull request by hadia**
# 📲 What

Remove butter knife from ThanksActivity and replaced with view binding

# 🤔 Why

deprecated butterknife

# 🛠 How

1- change ThanksActivity from java to kotlin
3-remove butter-knife and replace it with viewbinding

# 👀 See

https://user-images.githubusercontent.com/1075310/107649554-3ecd6e00-6c86-11eb-98d2-17aa699a469c.mp4

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA
Test Pledge project and after payment 
ThanksActivity Show

# Story 📖

https://kickstarter.atlassian.net/browse/NT-1764

### CODE REVIEW

**Review by Arkariang — APPROVED** [review]
Great! I tested it locally as well, great work 👍

File: app/src/main/java/com/kickstarter/ui/activities/ThanksActivity.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.concurrent.TimeUnit
+
+@RequiresActivityViewModel(ThanksViewModel.ViewModel::class)
+class ThanksActivity : BaseActivity<ThanksViewModel.ViewModel>() {
+    private lateinit var ksString: KSString
+    private lateinit var binding: ThanksLayoutBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ThanksLayoutBinding.inflate(layoutInflater)
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ui/activities/ThanksActivity.kt:33]: nit: If we could separate blocks of code with a whitespace similarly to the java code, that would help with readability.

File: app/src/main/java/com/kickstarter/ui/activities/ThanksActivity.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    private fun closeButtonClick() = viewModel.inputs.closeButtonClicked()
+
+    private fun showConfirmGamesNewsletterDialog() {
+        val optInDialogMessageString = ksString.format(
+            getString(R.string.profile_settings_newsletter_opt_in_message),
+            "newsletter",
+            getString(R.string.profile_settings_newsletter_games)
+        )
```
Comment (leighdouglas) [inline app/src/main/java/com/kickstarter/ui/activities/ThanksActivity.kt:99]: nit: whitespace after line 99

**Review by leighdouglas — APPROVED** [review]
LGTM! 🎉

**Pull request merged**